src/foam/u2/DateView.js

```diff
@@ -34,23 +34,39 @@ foam.CLASS({
     [ 'placeholder', 'yyyy-mm-dd' ]
   ],
 
+  listeners: [
+    {
+      name: 'onBlur',
```

Comment:
Let's try to explain why we need to do this. Here's a shot at it:

A slot's relateTo prevents feedback loops between the two slots that are related to eachother. This feedback prevention means that adapts and preSets on the Date properties won't be reflected in the UI. Consider the following example:
0. data has a preSet to ensure the date must be before today.
1. user selects a date in the future from DateView
2. relateTo's `inputToData` triggers
3. data's preSet changes the date from tomorrow to today
4. relateTo's `dataToInput` does NOT fire because `inputToData` is still in progress.

Triggering dataToInput onBlur ensures that the value being presented in the UI is the most up to date value because it avoids being boxed out by the feedback loop prevention.